sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -219,17 +230,44 @@ Connection getConnection() throws Exception {
       return toBuilder().setRowMapper(rowMapper).build();
     }
 
+    public Read<T> withCoder(Coder<T> coder) {
+      checkNotNull(coder, "coder");
+      return toBuilder().setCoder(coder).build();
+    }
+
     @Override
     public PCollection<T> apply(PBegin input) {
       return input
           .apply(Create.of(getQuery()))
-          .apply(ParDo.of(new ReadFn<>(this)));
+          .apply(ParDo.of(new ReadFn<>(this))).setCoder(getCoder())
+          // generate a random key followed by a GroupByKey and then ungroup
+          // to prevent fusion
+          // see https://cloud.google.com/dataflow/service/dataflow-service-desc#preventing-fusion
+          // for details
+          .apply(ParDo.of(new DoFn<T, KV<Integer, T>>() {
+            private int randInt;
+            @Setup
+            public void setup() {
+              Random random = new Random();
+              randInt = random.nextInt();
```

Comment:
My bad, I wanted to do new Random in @Setup, but nextInt in @ProcessElement. I'm fixing that.